StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TriviaHelper.cs

```diff
@@ -130,19 +133,30 @@ internal static SyntaxTriviaList WithoutLeadingWhitespace(this SyntaxTriviaList
         /// <param name="trivia">The trivia to create the list from.</param>
         /// <param name="triviaIndex">The index of the trivia in the created trivia list.</param>
         /// <returns>The created trivia list.</returns>
-        internal static SyntaxTriviaList GetContainingTriviaList(SyntaxTrivia trivia, out int triviaIndex)
+        internal static IReadOnlyList<SyntaxTrivia> GetContainingTriviaList(SyntaxTrivia trivia, out int triviaIndex)
         {
             var token = trivia.Token;
+            SyntaxTriviaList part1;
+            SyntaxTriviaList part2;
+
             triviaIndex = token.TrailingTrivia.IndexOf(trivia);
             if (triviaIndex != -1)
             {
                 var nextToken = token.GetNextToken(includeZeroWidth: true);
-                return token.TrailingTrivia.AddRange(nextToken.LeadingTrivia);
+
+                part1 = token.TrailingTrivia;
+                part2 = nextToken.LeadingTrivia;
+            }
+            else
+            {
+                var prevToken = token.GetPreviousToken();
+                triviaIndex = prevToken.TrailingTrivia.Count + token.LeadingTrivia.IndexOf(trivia);
```

Comment:
:exclamation: You cannot remote that. `triviaIndex` is an out parameter and several callers of the method require that value.